Symfony/CS/Tests/Fixer/Contrib/Php4ConstructorFixerTest.php

```diff
@@ -0,0 +1,541 @@
+     */
+    function __construct($bar)
+    {
+        var_dump(1);
+    }
+
+    function bar()
+    {
+        var_dump(3);
+    }
+}
+EOF;
+
+        $this->makeTest($expect, $input);
+    }
+
+    public function testPhp4Parent()
+    {
+        $input = <<<'EOF'
+<?php
+
+class Foo extends FooParent
+{
+    /**
+     * Constructor
+     */
+    function Foo($bar)
+    {
+        parent::FooParent(1);
+        var_dump(9);
+    }
+
+    function bar()
+    {
+        var_dump(3);
+    }
+}
+EOF;
+
+        $expect = <<<'EOF'
+<?php
+
+class Foo extends FooParent
+{
+    /**
+     * Constructor
+     */
+    function __construct($bar)
+    {
+        parent::__construct(1);
+        var_dump(9);
+    }
+
+    function bar()
+    {
+        var_dump(3);
+    }
+}
+EOF;
+
+        $this->makeTest($expect, $input);
+    }
+
+    public function testPhp4ParentInit()
+    {
+        $input = <<<'EOF'
+<?php
+
+class Foo extends FooParent
+{
+    /**
+     * Constructor
+     */
+    function Foo($bar)
+    {
+        parent::init(1);
+        var_dump(9);
+    }
+
+    function bar()
+    {
+        var_dump(3);
+    }
+}
+EOF;
+
+        $expect = <<<'EOF'
+<?php
+
+class Foo extends FooParent
+{
+    /**
+     * Constructor
+     */
+    function __construct($bar)
+    {
+        parent::init(1);
+        var_dump(9);
+    }
+
+    function bar()
+    {
+        var_dump(3);
+    }
+}
+EOF;
+
+        $this->makeTest($expect, $input);
+    }
+
+    public function testMixedParent()
+    {
+        $input = <<<'EOF'
+<?php
+
+class Foo extends FooParent
+{
+    /**
+     * Constructor
+     */
+    function __construct($bar)
+    {
+        parent::FooParent(1);
+        var_dump(9);
+    }
+
+    function bar()
+    {
+        var_dump(3);
+    }
+}
+EOF;
+
+        $expect = <<<'EOF'
+<?php
+
+class Foo extends FooParent
+{
+    /**
+     * Constructor
+     */
+    function __construct($bar)
+    {
+        parent::__construct(1);
+        var_dump(9);
+    }
+
+    function bar()
+    {
+        var_dump(3);
+    }
+}
+EOF;
+
+        $this->makeTest($expect, $input);
+    }
+
+    public function testParentOther()
+    {
+        $input = <<<'EOF'
+<?php
+
+class Foo extends FooParent
+{
+    /**
+     * Constructor
+     */
+    function Foo($bar)
+    {
+        $this->FooParent(1);
+        var_dump(9);
+    }
+
+    function bar()
+    {
+        var_dump(3);
+    }
+}
+EOF;
+
+        $expect = <<<'EOF'
+<?php
+
+class Foo extends FooParent
+{
+    /**
+     * Constructor
+     */
+    function __construct($bar)
+    {
+        parent::__construct(1);
+        var_dump(9);
+    }
+
+    function bar()
+    {
+        var_dump(3);
+    }
+}
+EOF;
+
+        $this->makeTest($expect, $input);
+    }
+
+    public function testClassWithAnonymous()
+    {
+        $input = <<<'EOF'
+<?php
+
+class Foo {
+    private $bar;
+
+    public function foo()
+    {
+        $this->bar = function () {};
+    }
+}
+EOF;
+        $expect = <<<'EOF'
```

Comment:
For consistency this should be expected. And expected should be before input (order of declaring like order of params for `makeTest`)